presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -593,11 +603,11 @@ public void addFinalTaskInfoListener(StateChangeListener<TaskInfo> stateChangeLi
 
     private synchronized void updateSplitQueueSpace()
     {
-        if (!whenSplitQueueHasSpaceThreshold.isPresent()) {
-            return;
-        }
-        splitQueueHasSpace = getQueuedPartitionedSplitCount() < whenSplitQueueHasSpaceThreshold.getAsInt();
-        if (splitQueueHasSpace) {
+        // Must check whether the unacknowledged split count threshold is reached even without listeners registered yet
```

Comment:
It might be a dumb question but I'm curious to understand why we need to check this before listeners are registered?

Reply:
It's possible for task initial splits to already exceed the pending split limit inside of the constructor, so setting the "no space" flag appropriately at initialization time had some effect but... now that I'm looking I can't see what the effect was. It might have had some relationship to the `MockRemoteTask` implementation in test and maybe isn't strictly required here.